### PR TITLE
[Autocomplete] Fix BC and trigger deprecation for `EntityAutocompleterInterface`

### LINKED ISSUE #793: [Autocomplete] potential BC break in the upcoming 2.x release
There is an, I think accidental, BC break in the 2.8-dev branch. Adding a new `EntityAutocompleterInterface::getGroupBy` method by https://github.com/symfony/ux/pull/481.

### PR DESCRIPTION
**Pull request by 1ed**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | -
| Tickets       | Fix #793
| License       | MIT

### CODE REVIEW

**Review by weaverryan — APPROVED** [review]
Thank you for catching that 👍  - I'll get this merged shortly

**Comment by weaverryan** [comment]
Thanks Gábor!

File: src/Autocomplete/src/AutocompleteResultsExecutor.php [hunk 1]
```diff
@@ -71,7 +71,9 @@ public function fetchResults(EntityAutocompleterInterface $autocompleter, string
 
         $results = [];
 
-        if (null === $groupBy = $autocompleter->getGroupBy()) {
+        if (!method_exists($autocompleter, 'getGroupBy')) {
+            trigger_deprecation('symfony/ux-autocomplete', '2.8', 'Not implementing the "EntityAutocompleterInterface::getGroupBy()" in "%s" is deprecated.', get_debug_type($autocompleter));
```
Comment (wouterj) [inline src/Autocomplete/src/AutocompleteResultsExecutor.php:75]: This deprecation is not needed, the `@method` annotation will already trigger a similar deprecation (which is done via Symfony's `DebugClassLoader`).

Comment (weaverryan) [reply src/Autocomplete/src/AutocompleteResultsExecutor.php:75]: thx - got it at https://github.com/symfony/ux/pull/808

**Pull request merged**